File: firstspringdemo/src/main/java/firstspring/firstspringdemo/DTOs/ProductDto.java
package firstspring.firstspringdemo.DTOs;

import java.util.UUID;

public class ProductDto {
    private UUID productId;
    private String name;
    private String description;
    private double price;
    private int quantity;
    private String imgUrl;
    private String locationId;
    
    public ProductDto() {
    }

    public ProductDto(UUID id, String name, String description, double price, int quantity) {
        this.productId = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.quantity = quantity;
    }

    public String getImgUrl() {
        return imgUrl;
    }
    public void setImgUrl(String imgUrl) {
        this.imgUrl = imgUrl;
    }
    public String getLocationId() {
        return locationId;
    }
    public void setLocationId(String locationId) {
        this.locationId = locationId;
    }
    
    public UUID getProductId() {
        return productId;
    }
    public void setProductId(UUID productId) {
        this.productId = productId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: firstspringdemo/flutter/lib/models/product_dto.dart
class ProductDto {
  final String? productId;
  final String? name;
  final String? description;
  final double? price;
  final int? quantity;
  final String? imgUrl;
  final String? locationId;

  ProductDto({
    this.productId,
    this.name,
    this.description,
    this.price,
    this.quantity,
    this.imgUrl,
    this.locationId,
  });

  factory ProductDto.fromJson(Map<String, dynamic> json) {
    return ProductDto(
      productId: json['productId'],
      name: json['name'],
      description: json['description'],
      price: json['price']?.toDouble(),
      quantity: json['quantity'],
      imgUrl: json['imgUrl'],
      locationId: json['locationId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'productId': productId,
      'name': name,
      'description': description,
      'price': price,
      'quantity': quantity,
      'imgUrl': imgUrl,
      'locationId': locationId,
    };
  }
}
